hal/halx86/acpi/busemul.c

```diff
@@ -14,8 +14,26 @@
 
 /* GLOBALS ********************************************************************/
 
+extern ULONG HalpShutdownContext;
+extern ULONG HalpPicVectorRedirect[HAL_PIC_VECTORS];
+
 /* PRIVATE FUNCTIONS **********************************************************/
 
+VOID
+NTAPI
+HalpCheckPowerButton(VOID)
+{
+    if ((!KiBugCheckData[0] && !InbvCheckDisplayOwnership()) ||
```

Comment:
What this function is expected to do?
(And since you've added the Inbv check, does that mean it is also supposed to display some text?)

Reply:
NT 5.2 ACPI Driver and Acpi Hal have an interface:
https://jira.reactos.org/browse/CORE-17378?focusedCommentId=127590&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-127590

When calling HaliAcpiMachineStateInit(), ACPI driver transmits ACPI HAL information, which is used including to fill HalpShutdownContext. It is used to call the HaliAcpiSleep() function under certain circumstances (for example, when HalpDisableHibernate is TRUE or HalpDisableS5Hibernation is TRUE).

ACPI HAL calls InbvAcquireDisplayOwnership() in two cases:
from HalHandleNMI(), which ends with halt;
from HalReturnToFirmware(), which then causes HalpReboot().

Thus, this function is performed when
if (KiBugCheckData[0] || InbvCheckDisplayOwnership()) &&
    HalpShutdownContext != 0)
{
   ...
}

This function reads and writes to registers
PM1a_EVT_BLK, PM1b_EVT_BLK, PM1a_CNT_BLK, PM1b_CNT_BLK (see Fixed ACPI Description Table (FADT) decription in ACPI specification).

Until there is an interface with the ACPI driver, it makes no sense to implement this function. It's just a trap for the future.